routing/taskana-spi-routing-dmn-router/src/test/java/pro/taskana/routing/dmn/DmnTaskRouterAccTest.java

```diff
@@ -2,29 +2,114 @@
-class DmnTaskRouterAccTest extends AbstractAccTest {
+@WithServiceProvider(
+    serviceProviderInterface = TaskRoutingProvider.class,
+    serviceProviders = DmnTaskRouter.class)
+@TaskanaIntegrationTest
+class DmnTaskRouterAccTest {
+  TaskanaConfiguration taskanaConfiguration;
+  @TaskanaInject TaskService taskService;
+  @TaskanaInject ClassificationService classificationService;
+  @TaskanaInject WorkbasketService workbasketService;
+  @TaskanaInject UserService userService;
+  @TaskanaInject TaskanaEngine taskanaEngine;
+  @TaskanaInject InternalTaskanaEngine internalTaskanaEngine;
 
-  private final TaskService taskService = taskanaEngine.getTaskService();
+  ClassificationSummary defaultClassificationSummary;
+  WorkbasketSummary defaultWorkbasketSummary;
+  ObjectReference defaultObjectReference;
+  WorkbasketSummary workbasketToBeRouted1;
+  WorkbasketSummary workbasketToBeRouted2;
+  WorkbasketSummary workbasketToBeRouted3;
+
+  @WithAccessId(user = "businessadmin")
+  @BeforeAll
+  void setup() throws Exception {
+    taskanaConfiguration = taskanaEngine.getConfiguration();
+    defaultClassificationSummary =
+        defaultTestClassification().buildAndStoreAsSummary(classificationService);
+    defaultWorkbasketSummary = defaultTestWorkbasket().buildAndStoreAsSummary(workbasketService);
+
+    workbasketToBeRouted1 =
+        createWorkbasketWithDomainA("GPK_KSC").buildAndStoreAsSummary(workbasketService);
+    workbasketToBeRouted2 =
+        createWorkbasketWithDomainA("GPK_KSC_1").buildAndStoreAsSummary(workbasketService);
+    workbasketToBeRouted3 =
+        createWorkbasketWithDomainA("GPK_KSC_2").buildAndStoreAsSummary(workbasketService);
+
+    WorkbasketAccessItemBuilder.newWorkbasketAccessItem()
+        .workbasketId(defaultWorkbasketSummary.getId())
+        .accessId("user-1-2")
+        .permission(WorkbasketPermission.OPEN)
+        .permission(WorkbasketPermission.READ)
+        .permission(WorkbasketPermission.APPEND)
+        .buildAndStore(workbasketService);
+
+    defaultObjectReference = defaultTestObjectReference().build();
+
+    UserBuilder.newUser()
+        .id("user-1-2")
+        .firstName("Max")
+        .lastName("Mustermann")
+        .longName("Long name of user-1-2")
+        .buildAndStore(userService);
+
+    Map<String, String> mutableProperties = new HashMap<>();
+    String dmnTableProperty = "taskana.routing.dmn";
+    String newPath = "/dmn-table.dmn";
+    mutableProperties.put(dmnTableProperty, newPath);
+    changeProperty("properties", taskanaConfiguration, mutableProperties);
+
+    List<TaskRoutingProvider> taskRoutingProviders = SpiLoader.load(TaskRoutingProvider.class);
+    for (TaskRoutingProvider taskRoutingProvider : taskRoutingProviders) {
+      taskRoutingProvider.initialize(taskanaEngine);
+    }
+    TaskRoutingManager taskRoutingManager = internalTaskanaEngine.getTaskRoutingManager();
+    changeProperty("taskRoutingProviders", taskRoutingManager, taskRoutingProviders);
```

Comment:
Could you try to avoid setting property here? Instead, getting existing providers above instead of loading them could help. Let's talk about the other solution :)